src/src/com/microsoft/aad/adal/BasicWebViewClient.java

```diff
@@ -183,24 +183,6 @@ public void run() {
                                 view.loadUrl(loadUrl, headers);
                             }
                         });
-                    } catch (IllegalArgumentException e) {
-                        Logger.e(TAG, "Argument exception", e.getMessage(),
-                                ADALError.ARGUMENT_EXCEPTION, e);
-                        // It should return error code and finish the
-                        // activity, so that onActivityResult implementation
-                        // returns errors to callback.
-                        Intent resultIntent = new Intent();
-                        resultIntent.putExtra(
-                                        AuthenticationConstants.Browser.RESPONSE_AUTHENTICATION_EXCEPTION,
-                                        e);
-                        if (mRequest != null) {
-                            resultIntent.putExtra(
-                                    AuthenticationConstants.Browser.RESPONSE_REQUEST_INFO,
-                                    mRequest);
-                        }
-                        sendResponse(
```

Comment:
Catching IllegalArgumentException is generally should be avoided - we are aiming for a crash in here... So instead of the leak we would see app crash now.

The Java Programming Language, by Gosling, Arnold, and Holmes: "Unchecked run-time exceptions represent conditions that, generally speaking, reflect errors in your program's logic and cannot be reasonably recovered from at run time."